### DISCUSSION TITLE
Playback link protection

### DISCUSSION THREAD

**Discussion by volkankoc** [bluenviron/mediamtx#1662] [Questions]
Hello
I want to prevent the rtsp camera url address that I have published from working on other computers. How can I do this with a token.

I did this with auth, but this time the user name and password must be entered each time. I don't want it to be this way. Or can this process be done in the background?

**Comment by volkankoc** [comment]
please help

**Comment by aler9** [comment]
Hello, you can either

* publish the stream to a path that already includes a token in its name:

   ```
   ffmeg -re -i ... -f rtsp rtsp://localhost:8554/mypath_token12313425234213
   ```
   
* If clients are reading the stream with HLS or WebRTC, you can write a reverse proxy that implements an authentication logic (cookie-based, token-based) with any language you like and place it in front of the server.

**Comment by volkankoc** [comment]
When I add a proxy to the frontend. The client will still be able to see the link from the source code while watching. That way he can still watch.

I wonder if I'm doing something wrong. Can you show a sample code or yml file?

**Comment by volkankoc** [comment]
It is true that if there is no user name and password in this system, he will not be able to enter. However, once logged in, the user will be able to get the player address in the source code of the website. Anyone who has this address will be able to watch the cameras. What can I do in this situation?

**Comment by aler9** [comment]
You are not talking about a reverse proxy, you are talking about placing an HTTP server near the media server and allowing direct traffic to the media server.

A reverse proxy is not a static HTTP server but a HTTP server that acts as proxy and filters incoming request directed to the server.

**Comment by volkankoc** [comment]
I understand So what settings do I need to make in the yml file? Will I redirect the reverse proxy externalAuthenticationURL through this setting?

**Comment by volkankoc** [comment]
Could you give more detailed information about this? How do I position the reverse proxy? externalAuthenticationURL should I use this parameter ?

[a comment by an automated account is not shown]
